### DISCUSSION TITLE
Version 6

### DISCUSSION THREAD

**Discussion by gilamran** [gilamran/tsc-watch#145] [General]
Here we'll be talking about the new version and its content.

For now you can install it using `npm install tsc-watch@dev` and give feedback.
Thanks

**Comment by pp0rtal** [comment]
Thanks @gilamran for this pre-release :+1: 

### Works as expected

- Enhancement relative to TSFILE / list/signal Emitted files behavior is as expected:
  - `--signalEmittedFiles --listEmittedFiles` shows `TSFILE:` statements
  - `--listEmittedFiles` shows `TSFILE:` statements too
  - `--signalEmittedFiles` is **hiding** `TSFILE:` as expected :)

- Enhancement relative to process error handling is as expected
  - I'm now able to see the **stderr** output of my tsc command, in my **stderr** of my parent process :relieved: 
  - I can also catch the `tsWatch.on("exit", (code, signal) => { ... })` and code / signal is transmitted
  - Type support works as expected and is improved (`file_emitted` + `exit` now exists in emitted types)
```
<--- Last few GCs --->
on[3214334:0x60b1b80]      818 ms: Mark-sweep (reduce) 50.2 (51.9) -> 49.5 (51.9) MB
...
```

**Comment by pp0rtal** [comment]
### Things to fix

Regression:
- TSFILE **black text** should have gone in v5, it's not the case of v6
  - meaning your fix 8ad7e7cf1e72fd665b6a0c8caa4b65abd5448b8c was actually missing in this branch 
  - suggested fix 4495ba114aec4db15f5af758b4a140a30d78f935

Major bug I introduced:

- `--listEmittedFiles` is not transmitted to tsc anymore (I tested above with a local fix), and for 2 reasons:
  - We need to push it **after** tsc binary (`node ./nod_modules/.bin/tsc --listEmittedFiles`) obviously and not `node --listEmittedFiles ./nod_modules/.bin/tsc ...`
  - As we are picking `--listEmittedFiles` we need to push it back whatever if `signalEmittedFiles` or `requestedToListEmittedFiles` are true
  - suggested fix 98e1746c987f49f08743d7e2b97d2ea0b3a9fec0

Uncommon thing:

- I have to import the client like this `import { TscWatchClient } from "tsc-watch/dist/client";`
  - The problem is that there is a default for tsc-watch main process in the package.json _(`"main": "./dist/client/index.js", "types": "./dist/client/index.d.ts"`)_
  - however `TscWatchClient` is not exposed
  - suggested fix cadf5cc9da21281fdc79bd5d8d39573aee35849b I suggest to create new files.
We can't expose what is in `lib/tsc-watch` & `lib/client` because `lib/tsc-watch` evals things but tell me if there is better
After this I can cleanly import the client `import { TscWatchClient } from "tsc-watch/client";`


PR https://github.com/gilamran/tsc-watch/pull/151

**Reply by gilamran** [reply]
I've merged the PR, I'll publish a new `dev` version later today

**Comment by pp0rtal** [comment]
@gilamran I did all my initial tests above again, Everything is working properly :tada: 

I'd say there are no regressions, as it's all the features added between v4 -> v5 support, Ts refactor must be safe.
I'm confident for a first v6 (confident enough to use it already in my work at least :).
Some doc is to update for `{ TscWatchClient }` and new `--signalEmittedFiles`, do I let you take care of this?

Thanks a lot for moving on this!

**Reply by pp0rtal** [reply]
Using the `6.0.0-dev.3` version for my usecase (client), still working great so far.

**Comment by gilamran** [comment]
ok, I'll publish version 6.0.0 this weekend. thanks for all the effort!
